skbeam/core/dpc.py

```diff
@@ -366,52 +357,45 @@ def dpc_runner(ref, image_sequence, start_point, pixel_size, focus_to_det,
         [(1, 5), (2, 6)] --> 2 bad pixels --> (1, 5) and (2, 6). Default is
         None.
 
-    negate : bool, optional
-        If True (default), negate the phase gradient along x direction before
-        reconstructing the final phase image. Default is True.
-
-    scale : bool, optional
-        If True, scale gx and gy according to the experiment set up.
-        If False, ignore pixel_size, focus_to_det, energy. Default is True.
-
-    Returns
-    -------
-    phase : ndarray
-        The final reconstructed phase image.
-
-    amplitude : ndarray
-        Amplitude of the sample transmission function.
+    Yields
+    ------
+    dpc_state : namedtuple
+        The internal state that `dpc_runner` requires for each iteration.
+        Can be passed to reconstruct_phase_from_partial_info which, along
+        with some additional info, will produce the final phase image
 
     References: text [1]_
     .. [1] Yan, H. et al. Quantitative x-ray phase imaging at the nanoscale by
     multilayer Laue lenses. Sci. Rep. 3, 1307; DOI:10.1038/srep01307 (2013).
 
     """
+    def initialize_state(scan_rows, scan_cols, ref, roi, bad_pixels):
```

Comment:
I don't know.  I had the same question.  The reason why I have it internal to lazy_dpc is because I can't think of why you would want to initialize the state yourself.  Seems like the only use would be to pass it in to the lazy_dpc function.